pallets/connectors-gateway/connectors-gateway-axelar-precompile/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::alloc::string::ToString;
+use ethabi::Token;
+use fp_evm::PrecompileHandle;
+use pallet_evm::{ExitError, PrecompileFailure};
+use precompile_utils::prelude::*;
+use sp_core::{bounded::BoundedVec, ConstU32, H160, H256, U256};
+use sp_runtime::{traits::Convert, DispatchResult};
+use sp_std::vec::Vec;
+
+pub const MAX_SOURCE_CHAIN_BYTES: u32 = 32;
+pub const MAX_SOURCE_ADDRESS_BYTES: u32 = 32;
+pub const MAX_TOKEN_SYMBOL_BYTES: u32 = 32;
+pub const MAX_PAYLOAD_BYTES: u32 = 32;
+pub const PREFIX_CONTRACT_CALL_APPROVED: [u8; 32] = keccak256!("contract-call-approved");
+
+pub type String<const U32: u32> = BoundedString<ConstU32<U32>>;
+pub type Bytes<const U32: u32> = BoundedBytes<ConstU32<U32>>;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	// Import various types used to declare pallet in scope.
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_core::H160;
+	use sp_std::vec::Vec;
+
+	// Simple declaration of the `Pallet` type. It is placeholder we use to
+	// implement traits and method.
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + pallet_evm::Config + pallet_connectors_gateway::Config
+	{
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
```

Comment:
Should we also add a type with a trait that the connectors gateway implements? This way we'll be able to mock that and also add some tests here and ensure that the right stuff is being call and so forth.

Reply:
We can not unfortunately, as we will need to call the extrinsic explictily. Or is it trait abstracted somwhere?